Sources/Fluent/Query/Codable/QueryDataDecoder.swift

```diff
@@ -1,25 +1,25 @@
-final class QueryDataDecoder<Database> where Database: QuerySupporting {
-    init(_ database: Database.Type) { }
-    func decode<D>(_ type: D.Type, from data: [QueryField: Database.QueryData]) throws -> D where D: Decodable {
-        let decoder = _QueryDataDecoder<Database>(data: data)
+final class QueryDataDecoder<M> where M: Model, M.Database: QuerySupporting {
+    init(_ model: M.Type) { }
+    func decode<D>(_ type: D.Type, from data: [QueryField: M.Database.QueryData]) throws -> D where D: Decodable {
+        let decoder = _QueryDataDecoder<M>(data: data)
         return try D.init(from: decoder)
     }
 }
 
 /// MARK: Private
 
-fileprivate final class _QueryDataDecoder<Database>: Decoder where Database: QuerySupporting {
+fileprivate final class _QueryDataDecoder<M>: Decoder where M: Model, M.Database: QuerySupporting {
```

Comment:
I think I'd prefer to keep this generic on the `Database` and just pass in `entity: String` during init so that this can be used for non-model types (very common) as well!